viewing_party/party.py

```diff
@@ -0,0 +1,168 @@
+
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    total = 0
+    for each in range(len(user_data["watched"])):
+        total += user_data["watched"][each]["rating"]
+    if user_data["watched"] == []:
+        return 0
+    avg = (total / (len(user_data["watched"])))
+    print (avg)
+    return avg
+
+def get_most_watched_genre(user_data):
+    counter_dict = {}
+    for movie_dict in user_data["watched"]:
+        if user_data["watched"] == []:
+            break
+        if counter_dict.get(movie_dict["genre"]):
+            counter_dict[movie_dict["genre"]] += 1
+        else:
+            counter_dict[movie_dict["genre"]] = 1
+    if not counter_dict:
+        return None
+    try:
+        most_watched_value = max(counter_dict, key=lambda x : x[1])
+    except ValueError:
+        return None
+    most_watched_genre = []
+    return most_watched_value
+
+def get_unique_watched(user_data):
+    seen = []
+    show_friends = []
+    for i in range(len(user_data["friends"])):
+        seen += user_data["friends"][i]["watched"]
+    for i in range(len(user_data["watched"])):
+        if user_data["watched"][i] not in seen:
+            show_friends.append(user_data["watched"][i])
+
+    if len(show_friends) == 0:
+        return []
+    return show_friends
+
+def get_friends_unique_watched(user_data):
+    seen = []
+    friends_show = []
+    for i in range(len(user_data["friends"])):
+        seen += user_data["friends"][i]["watched"]
+    no_repeats  = []
+    for i in friends_show:
+        if i not in no_repeats:
+            no_repeats.append(i)
+    friends_show = no_repeats
+    for each in seen:
+        if each not in user_data["watched"]:
+            friends_show.append(each)
+
+    if len(friends_show) == 0:
+        return []
+    no_repeats = []
+    for i in friends_show:
+        if i not in no_repeats:
+            no_repeats.append(i)
+    friends_show = no_repeats
+    return friends_show
+
+def get_available_recs(user_data):
+    seen = []
+    rec_list = []
+    for i in range(len(user_data["friends"])):
+        seen += user_data["friends"][i]["watched"]
+    user_seen = []
+    for element in user_data["watched"]:
+        user_seen.append(element["title"])
+    for each in seen:
+        if each["title"] not in user_seen:
+            rec_list.append(each)
+    print (rec_list)
+
+    subbed_only= []
+    for dict in rec_list:
+        if dict["host"] in user_data["subscriptions"]:
+            subbed_only.append(dict)
+    rec_list = subbed_only
+    if len(rec_list) == 0:
+        return []
+    no_repeats = []
+    for i in rec_list:
+        if i not in no_repeats:
+            no_repeats.append(i)
+    rec_list = no_repeats
+    return rec_list
+
+def get_new_rec_by_genre(user_data):
+    genre_counter = {}
+    rec_list = []
+    if user_data["watched"] == []:
+        return []
+    for list in user_data["watched"]:
+        if list["genre"] not in genre_counter.keys():
```

Comment:
```suggestion
        if list["genre"] not in genre_counter:
```

The `in` operator is actually an iterator under the hood. The default iterator for dictionaries are keys, so you can leave out `keys()`.